test/integration/apiserver/peerproxy/peer_proxy_test.go

```diff
@@ -177,7 +177,13 @@ func TestPeerProxiedRequestToThirdServerAfterFirstDies(t *testing.T) {
 
 	var jobsB *v1.JobList
 	// list jobs using ServerB which it should proxy to ServerC and get back valid response
-	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 1*time.Minute, false, func(ctx context.Context) (bool, error) {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+		default:
+		}
+
 		jobsB, err = kubeClientSetB.BatchV1().Jobs("default").List(context.Background(), metav1.ListOptions{})
 		if err != nil {
 			return false, nil
```

Comment:
add a log here so we can debug the test
```suggestion
                        t.Logf("error trying to list jobs from ServerB: %v", err)
			return false, nil
```

Reply:
same below, if jobsB== nil , log about it

Reply:
Done